diff --git a/src/pr_data.c b/src/pr_data.c
index ec51f02..3f0180f 100644
--- a/src/pr_data.c
+++ b/src/pr_data.c
@@ -13,6 +13,7 @@ THIS SOFTWARE IS PROVIDED BY Teemu Ikonen and Tom Schoonjans ''AS IS'' AND ANY E
 
 
 #include <stdio.h>
+#include <unistd.h>
 #include "xraylib.h"
 #include "xrayglob.h"
 
@@ -1219,5 +1220,7 @@ int main(void)
   SetErrorMessages(1);
   fclose(f);
 
+  sleep(1);
+
   return 0;
 }
